### ISSUE TITLE
Something seriously wrong.

### ISSUE THREAD

**Issue by mrbid** [minetest-mods/i3#56]
There is something seriously wrong here, I have extensively tested i3 on a world with no other mods and I am still able to trigger this strange bug at what seems like completely random interactions with other mods. Other players experience it online, so it's not just my client, it's baffling. It seems to work fine on some servers with particular mods but there's something that i3 is doing that I cant put my finger on, that when combined with other mods in some way, causes it to just completely break formspecs of every other mod in the game.

I run a number of servers, I have only got i3 working on one of them.

It made me think it was due to the order i3 was loaded in with other mods, but really, I have no idea at this point.

It is a shame because i3 is a nice inventory, but I just can't use it, all these problems vanish across all servers when I remove i3.

i3 is 100% absolutely breaking formspecs for all mods including the minetest_game under circumstances that I cannot isolate, I cannot work out whats causing it, why it seems to happen randomly in certain conditions where other mods are installed on the server, I have tried so many combinations, i3 just seems to randomly decide to break formspecs or not break them and there is no logical root to figuring out why. Something is very wrong.

**Comment by SmallJoker** [comment]
* Please provide screenshots
* Re-test in vanilla MTG + i3, both up-to-date and with the newest MInetest build you can find
* Provide exact reproduction steps
* Narrow down the issue to specific client versions

i3 does not make use of formspec prepends, hence has no influence whatsoever to other mods.

**Comment by mrbid** [comment]
I have tried so many combinations believe me, and I cannot escape the bug.

All I can suggest is you clone my worldmods and see for yourself.

https://github.com/mrbid/Minetest-Worldmods/

Everything is running on 5.5.0

**Comment by kilbith** [comment]
Please follow the issue guidelines described by @SmallJoker

I'm yet to see anything helpful in this issue.

**Comment by kilbith** [comment]
Forks are not accounted as bug reports here.

Also, tl;dr what are the “bugs” essentially ?

**Comment by mrbid** [comment]
Well the bug seems to be that all other formspecs break, on pretty much every other mod I have tested that uses a formspec it seems their on_receive_fields() function stops working has the formspec will load but not respond to pressing buttons, hence why I think the chest no longer exists the close animation as I am able to open the fromspec and escape out of it but there is no signal sent from the closing of the formspec to tell the chest to go back to it's closed state.

**Comment by kilbith** [comment]
There is no connection between a node's formspec (= map metadata) and an inventory formspec (player only).

**Comment by mrbid** [comment]
I think the problem may have been that in my frustration and confusion when trying to denominate down which part of the code was causing the issue file by file.

**Comment by mrbid** [comment]
I think maybe this bug was a change I made, I think that change was in `core.register_on_player_receive_fields` in `fields.lua` and when I commented out:
```
if formname == "i3_outdated" then
return false, core.kick_player(name,
	S"Come back when your Minetest client is up-to-date (www.minetest.net).")
elseif formname ~= "" then
return false
end
```

I should have left behind:
`if formname ~= "" then return false end`

I could see how that could break every formspec, but, what is strange is that this still worked on one of my servers, and when testing this was some what of a schrodinger's cat situation. I wonder why, I wonder why it didn't just stay broken like this, why it seemed to work at time and not other times? If this is the route cause, and it certainly seems culpable.
